v2/contribute/decisions/oauth/0004-scope-list.mdx

```diff
@@ -0,0 +1,39 @@
+---
+id: "0004"
+title: Store a global list of scopes that can be enabled per client
+hide_title: true
+---
```

Comment:
The standard says:
```
   If the client omits the scope parameter when requesting
   authorization, the authorization server MUST either process the
   request using a pre-defined default value or fail the request
   indicating an invalid scope.
```
Based on this, I suggest defaulting to an `openid` scope. In this case the `id_token` should only contain the default JWT claims (sub, iat, exp, iss, aud) but no other info like email or phone number.

Reply:
Right. This makes sense. @AdithyaSSup please take a note of this since this will affect the auth code creation endpoint in the core. Also, `openid` is a reserved scope that doesn't start with `supertokens`  - so we need to prevent users from deleting this scope.

Reply:
1. I don't think we even need to add it to the allowed scopes for client table, since this is always allowed and cannot be deleted.
2. No, we don't really need to add it to the scope list if it exists; if the client is not requesting an id_token we should not generate one.

Reply:
@rishabhpoddar .. As @porcellus suggested, we can have `openid` as an implicit default scope. Then in such a case ,there isn't any need to store/delete, and if the user tries to remove this scope from the required list of scopes , then such a request can be invalidated.

However , I have few questions pertaining to addition of default scope.
- if the user tries add `openid` to the list of required scopes during client creation, should we notify the user that scope `openid` is a default scope and cannot be added or , should  we just allow client creation ignoring the `openid` scope?.
- Also, since `openid` default implicit scope, should we consider showing in the allowed list of scopes?

Reply:
> I don't think we even need to add it to the allowed scopes for client table

I think it's easier from a logic / db point of view to just add it always in the allowed scopes for each client.

> No, we don't really need to add it to the scope list if it exists; if the client is not requesting an id_token we should not generate one.

So if during the auth code grant flow, the scopes is an empty array, should we throw an error?

> if the user tries to add openid to the list of required scopes during client creation, should we notify the user that scope openid is a default scope with a warning or a note or , should we just allow client creation ignoring the openid scope?.

If the openid scope is a part of the allowed scopes array during client creation, we keep it in there. If it's not, we will add it. Also, instead of "list of required scopes", did you mean "list of allowed scopes"?

> Also, since openidis an implicit default scope, should we consider showing in the allowed list of scopes?
What do you mean?

Reply:
> So if during the auth code grant flow, the scopes is an empty array, should we throw an error

No, but we should only add it in the empty array/no array case.

Reply:
> Also, instead of "list of required scopes", did you mean "list of allowed scopes"?

No, I meant, if the user doesn't explicitly add `openid` scope during client creation, then in such a case should we display `openid` as one of the scopes in the create client api response?

Reply:
> No, but we should only add it in the empty array/no array case.

I don't think so. If the scopes array is empty during the auth code grant flow, we should just keep it empty and NOT issue the id token. The access token will contain minimum info (just the user ID and start jwt claims). 

What do you think?

---------
Just to clarify (since this conversation is confusing now), ONLY if a CLIENT is created with an empty allowed scopes array, we automatically add the openid scope to their allowed scopes.

Reply:
This is what I actually meant, not sure how I missed the auth code grant flow part.

Reply:
Okay. Please add to some ADR. Then i will close this comment